python/ray/experimental/workflow/execution.py

```diff
@@ -0,0 +1,100 @@
+import logging
+import time
+from typing import Union, Optional
+
+import ray
+
+from ray.experimental.workflow import workflow_context
+from ray.experimental.workflow import recovery
+from ray.experimental.workflow.common import Workflow
+from ray.experimental.workflow.step_executor import (commit_step,
+                                                     execute_workflow)
+from ray.experimental.workflow.storage import (
+    Storage, create_storage, get_global_storage, set_global_storage)
+from ray.experimental.workflow.workflow_access import flatten_workflow_output
+
+logger = logging.getLogger(__name__)
+
+
+def run(entry_workflow: Workflow,
+        storage: Optional[Union[str, Storage]] = None,
+        workflow_id: Optional[str] = None,
+        detached: bool = True) -> ray.ObjectRef:
+    """
+    Run a workflow asynchronously.
+
+    Args:
+        entry_workflow: The workflow to run.
+        storage: The storage or the URL of an external storage used for
+            checkpointing.
+        workflow_id: The ID of the workflow. The ID is used to identify
+            the workflow.
+        detached: If True, the execution of workflow is detached form
+            the driver. Otherwise the workflow shares fate with the driver
+            (i.e. if the driver failed, the workflow also failed).
+
+    Returns:
+        The execution result of the workflow, represented by Ray ObjectRef.
```

Comment:
Docstring is duplicated from api.py, can we remove (or just reference the other)?